shared-libs/rules-engine/src/pouchdb-provider.js

```diff
@@ -44,13 +44,15 @@ const medicPouchProvider = db => {
 
     stateChangeCallback: docUpdateClosure(db),
 
-    commitTargetDoc: (assign, userDoc, docTag) => {
-      const userContactId = userDoc && userDoc._id;
-      const _id = `target-${docTag}-${userContactId}`;
+    commitTargetDoc: (assign, userContactDoc, userSettingsDoc, docTag) => {
+      const userContactId = userContactDoc && userContactDoc._id;
+      const userSettingsId = userSettingsDoc && userSettingsDoc._id;
+      const _id = `target-${docTag}-${userContactId}-${userSettingsId}`;
       const createNew = () => ({
         _id,
         type: 'target',
-        user: userContactId,
+        user: userSettingsId,
```

Comment:
if the target doc's `user` attriute is not used for replication, is it still needed?